react-interactive/src/Menue/Application.tsx

```diff
@@ -43,7 +43,8 @@ interface IProps {
     AllowCollapsed?: boolean
     NavBarContent?: React.ReactNode,
     HideSideBar?: boolean,
-    UseLegacyNavigation?: boolean
+    UseLegacyNavigation?: boolean,
+    NavBarCallBack?: (div: HTMLDivElement) => void
```

Comment:
This callback would be used to access properties of the navbar like offsetWidth, offsetHeight, etc..

Reply:
so what triggers the callback? On every Render?

Reply:
It is triggered inside a layoutEffect so after all the DOM modifications.

Reply:
not sure a `setNavBarRef` is much better.
https://react.dev/reference/react/forwardRef
I'll put in a suggestion in a minute

Reply:
Added typing to fowardRef and defined children as a prop since we are now wrapping the components in fowardRef which doesn't explicitly include children as a prop.